### PR TITLE
Fetch and expand sonar analyzer

### PR DESCRIPTION
**Pull request by frederic-tingaud-sonarsource**

### CODE REVIEW

File: update_compilers/install_nonfree_compilers.sh [hunk 1]
```diff
@@ -116,3 +116,7 @@ install_cuda https://developer.nvidia.com/compute/cuda/10.1/Prod/local_installer
 install_cuda https://developer.nvidia.com/compute/cuda/10.1/Prod/local_installers/cuda_10.1.168_418.67_linux.run 10.1.168
 install_cuda https://developer.download.nvidia.com/compute/cuda/10.1/Prod/local_installers/cuda_10.1.243_418.87.00_linux.run 10.1.243
 install_cuda https://developer.download.nvidia.com/compute/cuda/10.2/Prod/local_installers/cuda_10.2.89_440.33.01_linux.run 10.2.89
+
+##################################
+# Sonar Analyzer
+s3fetch ${S3URL}/sonar/sonar.txz | tar Jxf -
```
Comment (mattgodbolt) [inline update_compilers/install_nonfree_compilers.sh:122]: Looks good! We will eventually need to come up with a `ce_install` approach for this but that's not important right now :) Thanks!

**Review by mattgodbolt — APPROVED** [review]

**Comment by mattgodbolt** [comment]
Unfortunately the tarball that's extracted doesn't have execute rights for the `sonar.sh` -

```
Mar/10 03:15 admin-node~ $ /opt/compiler-explorer/sonar/sonar.sh
fish: The file '/opt/compiler-explorer/sonar/sonar.sh' is not executable by this user
Mar/10 03:15 admin-node~ $ ls -l /opt/compiler-explorer/sonar/sonar.sh
-rw-r--r-- 1 1001 docker 161 Mar  8 21:52 /opt/compiler-explorer/sonar/sonar.sh
```

I can manually chmod +x it for now, but the next time our installer runs it will lose that ability.

**Comment by mattgodbolt** [comment]
Also:

```
Mar/10 03:17 admin-node~ $ bash /opt/compiler-explorer/sonar/sonar.sh
/opt/compiler-explorer/sonar/sonar.sh: line 3: java: command not found
```

there's no java on the path. the script edits `path` not `PATH` -- so I think that needs fixing too...

**Comment by mattgodbolt** [comment]
Oh, no that's the java path. We don't have `java` around on our machines: we will have to pick one of the many java runtimes to somehow use, maybe `/opt/compiler-explorer/jdk-18.0.1/bin/java` ? Our `ce_install`-based installers can have dependencies between things and rewrite scripts but for now I will edit the script

**Comment by mattgodbolt** [comment]
It's alive!

```
$ /opt/compiler-explorer/sonar/sonar.sh --help
{
  "header": "This is a preview of the static analysis capabilities of the Sonar C and C++ analyzer.\nTo experience the full capabilities of the Sonar solution, you can try the open-source and free\nSonarLint plugin from your IDE marketplace. You can also analyze your open source projects for\nfree with SonarCloud (https://sonarcloud.io/​) or use the Developer Edition of\nSonarQube (https://www.sonarqube.org/downloads/​).\nTo get more information including support for a wider list of languages and compilers,\ncheck out our website https://sonarsource.com​.\n",
  "parsingErrors": [],
  "issues": [],
  "logs": [
    {
      "level": "ERROR",
      "message": "Analysis failed: Unrecognized command line option \"--help\"."
    }
  ]
}
Mar/10 03:20 admin-node~ 4.5s
```

took 4.5s to start up though :)

**Comment by Godin** [comment]
> Unfortunately the tarball that's extracted doesn't have execute rights for the sonar.sh

> we will have to pick one of the many java runtimes to somehow use, maybe /opt/compiler-explorer/jdk-18.0.1/bin/java ?

@mattgodbolt  You're right, my mistake - we discussed both points with @frederic-tingaud-sonarsource , but I forgot to fix them.

**Comment by Godin** [comment]
@mattgodbolt I just uploaded archive with fixes - script is executable and uses `/opt/compiler-explorer/jdk-17.0.0/bin/java`

**Comment by mattgodbolt** [comment]
Fantastic! I'll aim to get this rolled out later today! Thanks!

**Pull request merged**